src/main/java/org/opensearch/jdbc/transport/http/ApacheHttpTransport.java

```diff
@@ -239,8 +247,12 @@ private CloseableHttpResponse doGet(URI uri, Header[] headers, int readTimeout)
             HttpGet request = new HttpGet(uri);
             request.setHeaders(headers);
             request.setConfig(getRequestConfig());
+
+            if (this.preemptiveCreds != null) {
+                request.addHeader(new BasicScheme().authenticate(this.preemptiveCreds, request, null));
+            }
             return httpClient.execute(request);
-        } catch (IOException e) {
+        } catch (IOException | AuthenticationException e) {
```

Comment:
What was happening without catching `AuthenticationException`?

Reply:
Compilation error because of `new BasicScheme().authenticate()`.